OsmAnd/src/net/osmand/plus/views/Renderable.java

```diff
@@ -0,0 +1,480 @@
+                        if (broken) {
+                            path.moveTo(x, y);
+                            broken = false;
+                        } else
+                            path.lineTo(x, y);
+                        lastx = x;
+                        lasty = y;
+                    } else
+                        broken = true;
+                } else
+                    broken = true;
+
+            }
+
+            canvas.drawPath(path, p);
+            canvas.rotate(tileBox.getRotate(), tileBox.getCenterPixelX(), tileBox.getCenterPixelY());
+
+            p.setStrokeWidth(pSw);
+            p.setColor(pCol);
+        }
+    }
+
+    //----------------------------------------------------------------------------------------------
+
+    public static class RenderableDot extends RenderableSegment {
+
+        float dotScale;
+
+        public RenderableDot(RenderType type, List<GPXUtilities.WptPt> pt, double param1, double param2) {
+            super(type, pt, param1, param2);
+
+            dotScale = (float) param2;
+        }
+
+        @Override public void recalculateRenderScale(OsmandMapTileView view) {
+            if (culled == null && culler == null) {
+                culler = new AsynchronousResampler(renderType, view, this, param1, param2);
+                assert (culler != null);
+                if (culler != null)
+                    culler.execute("");
+            }
+        }
+
+        private String getLabel(double value) {
+            String lab;
+            lab = String.format("%.2f km",value/1000.);
+            return lab;
+        }
+
+        @Override public void drawSingleSegment(Paint p, Canvas canvas, RotatedTileBox tileBox) {
+
+            assert p != null;
+            assert canvas != null;
+            assert tileBox != null;
+
+            try {
+
+                if (culled == null)
+                    return;
+
+                Paint px = new Paint();
+                assert (px != null);
+
+                px.setStrokeCap(Paint.Cap.ROUND);
+
+                canvas.rotate(-tileBox.getRotate(), tileBox.getCenterPixelX(), tileBox.getCenterPixelY());
+
+                float sw = p.getStrokeWidth();
+                float ds = sw * dotScale;
+
+                int w = tileBox.getPixWidth();
+                int h = tileBox.getPixHeight();
+
+                for (GPXUtilities.WptPt pt : culled) {
+
+                    float x = tileBox.getPixXFromLatLon(pt.lat, pt.lon);
+                    float y = tileBox.getPixYFromLatLon(pt.lat, pt.lon);
+
+                    if (isIn(x, y, w, h)) {
+
+                        px.setColor(0xFF000000);
+                        px.setStrokeWidth(ds + 2);
+                        canvas.drawPoint(x, y, px);
+                        px.setStrokeWidth(ds);
+                        px.setColor(0xFFFFFFFF);
+                        canvas.drawPoint(x, y, px);
+
+                        //TODO: I do not know how to correctly handle screen density!
```

Comment:
There is method view.getScaleCoefficient it is better to use it because it is also takes into account tablets (we enlarge for tablets 1.5 times everything what's visible on the map screen)